app/controllers/CancellationReasonController.scala

```diff
@@ -62,8 +67,21 @@ class CancellationReasonController @Inject() (
           value =>
             for {
               updatedAnswers <- Future.fromTry(request.userAnswers.set(CancellationReasonPage, value))
-              _              <- sessionRepository.set(updatedAnswers)
-            } yield Redirect(navigator.nextPage(CancellationReasonPage, updatedAnswers, departureId))
+              ie015Data <-
+                departureMessageService.getIE015FromDeclarationMessage(departureId)
+              ie014Data = IE015Data.fromIE015Data(ie015Data, value)
```

Comment:
Similar thing here to above with the option. If ie015Data was not optional we could remove the if case below